src/Repository/EventRepository.php

```diff
@@ -403,13 +403,15 @@ public function searchAllEvents(SearchParametersFilter $search): array
 LEFT JOIN adherents ON adherents.id = events.organizer_id
 LEFT JOIN committees ON committees.id = events.committee_id
 LEFT JOIN events_categories AS event_category ON event_category.id = events.category_id
+LEFT JOIN citizen_action_categories AS citizen_action_category ON citizen_action_category.id = events.category_id
 WHERE (events.address_latitude IS NOT NULL 
     AND events.address_longitude IS NOT NULL 
     AND (6371 * ACOS(COS(RADIANS(:latitude)) * COS(RADIANS(events.address_latitude)) * COS(RADIANS(events.address_longitude) - RADIANS(:longitude)) + SIN(RADIANS(:latitude)) * SIN(RADIANS(events.address_latitude)))) < :distance_max 
     AND events.begin_at > :today 
     AND events.published = :published
     AND events.status = :scheduled
-    AND event_category.status = :enabled)
+    AND (event_category.status = :enabled OR citizen_action_category.status = :enabled)
```

Comment:
This one was very tricky(long?) to debug :face_with_head_bandage: 

We aren't filtering citizen actions properly in production right now.

In short:
- the `events` table holds the `Event` and `CitizenAction` entity.
- `Event` entity has a ManyToOne relationship with `EventCategory`, through the `category` property.
- `CitizenAction` entity has a ManyToOne relationship with `CitizenActionCategory`, also through the `category` property.
- this method (`EventRepository::searchAllEvents()`) is meant to search through Events and CitizenActions at the same time.
- but the "category status" is filtered with a LEFT JOIN on EventCategories only.

Basically, our testsuite was passing because we were lucky that previous tests already loaded EventCategories with ids that are matching the CitizenActionCagories ids.


- cf [this test](https://github.com/EnMarche/en-marche.fr/blob/testsuite/tests/Repository/EventRepositoryTest.php#L50): 7 is including Citizen Actions.
- and cf [this test](https://github.com/EnMarche/en-marche.fr/blob/testsuite/tests/Controller/EnMarche/SearchControllerTest.php#L81): 6 is also including Citizen Actions.

Reply:
fixed :man_dancing:

Reply:
:yeah: !